### PR TITLE
age wise deceased chart added

### PR DESCRIPTION
**Pull request by strikertanmay**
**Description of PR**

**Type of PR**

- [ ] Bugfix
- [x] New feature

**Relevant Issues**  
Fixes #...

**Checklist**

- [x] Compiles and passes lint tests
- [x] Properly formatted
- [x] Tested on desktop
- [x] Tested on phone

**Screenshots**

### CODE REVIEW

File: src/components/Charts/deceasedchart.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  defaults.global.tooltips.intersect = false;
+  defaults.global.tooltips.mode = 'nearest';
+  defaults.global.tooltips.position = 'nearest';
+
+  defaults.global.legend.display = true;
+  defaults.global.legend.position = 'bottom';
+
+  defaults.global.hover.intersect = false;
+
+  const ages = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
```
Comment (iDhruv22) [inline src/components/Charts/deceasedchart.js]: Can be - 
```js
Array(10).fill(0)

File: src/components/Charts/deceasedchart.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  }
+
+  props.data.forEach((patient) => {
+    let flag = false;
+    if (patient.patientstatus === 'Deceased') {
+      flag = true;
+    }
+    if (patient.agebracket) {
+      const age = parseInt(patient.agebracket);
+      if (age >= 0 && age <= 10 && flag) {
```
Comment (iDhruv22) [inline src/components/Charts/deceasedchart.js]: Can be -
```js
for (let i = 0; i < 10; i++) {
  if (age > (i * 10) && age <= ((i + 1) * 10) && flag) {
    ages[i]++;
  }
}

File: src/components/Charts/deceasedchart.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+  defaults.global.hover.intersect = false;
+
+  const ages = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+  let unknown = 0;
+  if (!props.data || props.data.length === 0) {
+    return <div></div>;
+  }
+
+  props.data.forEach((patient) => {
```
Comment (iDhruv22) [inline src/components/Charts/deceasedchart.js:30]: All these operations can be wrapped under _useCallback_.

**Review by iDhruv22 — COMMENTED** [review]
Looks great - just minor good to have.

File: src/components/Charts/deceasedchart.js (hunk 3 above)
Comment (strikertanmay) [reply src/components/Charts/deceasedchart.js:30]: I think there is no such necessary need to wrap these operations under useCallback.

**Comment by saurabhsharma1976** [comment]
if deployed already perhaps title "PATIENT AGE" not clear - is it for mortality cases OR for tested cases?

**Comment by strikertanmay** [comment]
PATIENT AGE is for age wise positive cases.
I made a PR for Age wise deceased cases

**Comment by saurabhsharma1976** [comment]
thanks

**Comment by sudevschiz** [comment]
This should probably be done as a toggle in the same Age-wise distribution graph, so that user can switch between different criteria. What do you think?

**Comment by saurabhsharma1976** [comment]
great idea to put up self serve option!

**Comment by strikertanmay** [comment]
> This should probably be done as a toggle in the same Age-wise distribution graph, so that user can switch between different criteria. What do you think?

I think, for now, we can just add one other graph-like others. In the future, we will bring this toggling feature and other features through which you can interact with graphs.

**Comment by saurabhsharma1976** [comment]
Thanks. Best to keep it iterative and if needed

Rgds
Saurabh

________________________________

**Comment by jeremyphilemon** [comment]
Hey, thanks for the PR! 

Please raise PRs against issues you've been assigned to and have them referenced in the PR :)

**Pull request closed without merging**